core/src/main/java/io/aiven/klaw/service/AclControllerService.java

```diff
@@ -905,13 +939,42 @@ private void updateServiceAccountsForTeam(AclRequests aclRequest, int tenantId)
         serviceAccounts.getServiceAccountsList().add(aclRequest.getAcl_ssl());
         optionalTeam.get().setServiceAccounts(serviceAccounts);
       }
+
       // Update team with service account
       manageDatabase.getHandleDbRequests().updateTeam(optionalTeam.get());
       commonUtilsService.updateMetadata(
           tenantId, EntityType.TEAM, MetadataOperationType.UPDATE, null);
     }
   }
 
+  private void removeServiceAccountOnTransferOfOwnership(AclRequests aclRequest, int tenantId) {
+    if (Objects.equals(RequestOperationType.CLAIM.value, aclRequest.getRequestOperationType())) {
+      if (manageDatabase
```

Comment:
how about moving this condition to a higher level ? So transfer (remove/add) only for this condition.

if (!manageDatabase
          .getHandleDbRequests()
          .existsAclSslInTeam(
              aclRequest.getTeamId(), aclRequest.getTenantId(), aclRequest.getAcl_ssl())) {
        // Team still has other Acls left with service user so do not remove from here.
        transferServiceUserOwnership(aclReq);
      }